src/app/core/services/filters.service.ts

```diff
@@ -365,4 +378,92 @@ export class FiltersService {
     const milestones = this.milestoneService.milestones;
     return [...milestones, Milestone.PRWithoutMilestone, Milestone.IssueWithoutMilestone];
   }
+
+  private getGhFilterDeselectedLabels(deselectedLabels: Set<string>): string {
+    return Array.from(deselectedLabels)
+      .map((label) => BooleanConjunctions.EXCLUDE + FilterOptions.label + `\"${label}\"`)
+      .join(BooleanConjunctions.AND);
+  }
+
+  private getGhFilterLabels(labels: string[]): string {
+    return labels.map((label) => FilterOptions.label + `\"${label}\"`).join(BooleanConjunctions.AND);
+  }
+
+  private getGhFilterMilestones(milestones: string[]): string {
+    return milestones
+      .map((milestone) => (MilestoneFilter.hasOwnProperty(milestone) ? MilestoneFilter[milestone] : FilterOptions.milestone + milestone))
+      .join(BooleanConjunctions.OR);
+  }
+
+  private getGhFilterSort(sort: Sort): string {
+    return SortFilter.hasOwnProperty(sort.active) ? SortFilter[sort.active] + ':' + sort.direction : '';
+  }
+
+  private getGhFilterTypes(type: string): string {
+    return TypeFilter[type];
+  }
+
+  /**
+   * Returns the encoded filter string for the GitHub search using url queries
+   * @returns Encoded filter string
+   */
+  getEncodedFilter(): string {
+    const res = [
+      '',
+      this.getGhFilterDeselectedLabels(this.filter$.value.deselectedLabels),
+      this.getGhFilterLabels(this.filter$.value.labels),
+      this.getGhFilterMilestones(this.filter$.value.milestones),
+      this.getGhFilterSort(this.filter$.value.sort),
+      this.getGhFilterTypes(this.filter$.value.type),
+      this.getGhFilterOpenAndClosedPR(this.filter$.value.assignees, this.filter$.value.status),
+      this.filter$.value.title
```

Comment:
Let's add a comment to link to [#425 ](https://github.com/CATcher-org/WATcher/issues/425) so that future devs are aware of this discrepency.